src/main/java/org/ladocuploader/app/utils/HouseholdUtilities.java

```diff
@@ -1,34 +1,82 @@
+
+  public static Calendar ECE_CUTOFF_DATE = new Calendar.Builder().setDate(2019, 9, 30).build();
+  /**
+   * Checks the birthdate against the current calendar date (now) to see if the birthdate is 18 years or older.
+   * @param day day of Birth
+   * @param month month of Birth
+   * @param year year of Birth
+   * @return boolean - true if member is 18 or older, otherwise false
+   * @throws NumberFormatException throws this if the day, month or year passed in are invalid
+   */
+  public static boolean isMember18orOlder(int year, int month, int day) throws NumberFormatException {
+    return isMember18orOlder(year, month, day, Calendar.getInstance());
+  }
+  /**
+   * Checks the birthdate against the calendar date passed in to see if the birthdate is 18 years old or older.
+   *
+   * @param day day of Birth
+   * @param month month of Birth
+   * @param year year of Birth
+   * @param calendar calendar to compare the birthdate to
+   * @return boolean - true if member is 18 or older, otherwise false
+   * @throws NumberFormatException throws this if the day, month or year passed in are invalid
+   */
+  public static boolean isMember18orOlder(int year, int month, int day, Calendar calendar) throws NumberFormatException {
 
     if (day <= 0 || month <= 0 || year <= 0 ) {
       throw new NumberFormatException("cannot analyze birthdate as fields are missing");
     }
 
-    Calendar memberBirthDayCal = Calendar.getInstance();
-    memberBirthDayCal.set(Calendar.YEAR, year);
-    memberBirthDayCal.set(Calendar.MONTH, month);
-    memberBirthDayCal.set(Calendar.DAY_OF_MONTH, day);
+    Calendar memberBirthDayCal = new Calendar.Builder().setDate(year, month, day).build();
 
-    Calendar cal = Calendar.getInstance();
-    cal.set(year, month, day);
+    Calendar cal = new Calendar.Builder()
+            .setDate(calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH), calendar.get(Calendar.DAY_OF_MONTH)).build();
     cal.add(Calendar.YEAR, -18);
 
     // these are converted to milliseconds since Epoch and then compared.
     // if the memberBirthDayCal is < or == the cal, then they are 18+ years old.
     return memberBirthDayCal.compareTo(cal) <= 0;
   }
 
+  /**
+   * Checks to see if the member is eligible for ECE, based on if the member is pregnant or if they are young enough.
+   *
+   * @param member household member data
+   * @param inputData full input data for the submission
+   * @return boolean - true if member is eligible, false if not
+   * @throws NumberFormatException throws this if the day, month or year passed in are invalid
+   */
+  public static boolean isMemberEceEligible(Map<String, Object> member, Map<String, Object> inputData) throws NumberFormatException {
+    boolean isPregnant = ((List)inputData.getOrDefault("pregancies[]", List.of())).contains(member.get("uuid"));
+
+    if (isPregnant) {
+      return true;
+    }
+
+    int birthDay = Integer.parseInt((String) member.get("householdMemberBirthDay"));
```

Comment:
Does `parseInt` work if DOB is not provided? I can't remember if this is a required field or not

Reply:
Will ask Roshan about this in the morning, and what should happen if it's not provided

Reply:
Ignoring this as we are making these fields required as part of another ticket